From 5ae1fe0156581c64ac0c587aef7a23d6a99987a4 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Fri, 9 Nov 2018 01:17:27 -0500
Subject: [PATCH 11/12] Monumenta - Disable phantom spawning

Signed-off-by: Byron Marohn <combustible@live.com>
---
 .../net/minecraft/server/MobSpawnerPhantom.java    | 60 +---------------------
 1 file changed, 1 insertion(+), 59 deletions(-)

diff --git a/src/main/java/net/minecraft/server/MobSpawnerPhantom.java b/src/main/java/net/minecraft/server/MobSpawnerPhantom.java
index 0626b779..270a44c5 100644
--- a/src/main/java/net/minecraft/server/MobSpawnerPhantom.java
+++ b/src/main/java/net/minecraft/server/MobSpawnerPhantom.java
@@ -10,64 +10,6 @@ public class MobSpawnerPhantom {
     public MobSpawnerPhantom() {}
 
     public int a(World world, boolean flag, boolean flag1) {
-        if (!flag) {
-            return 0;
-        } else {
-            Random random = world.random;
-
-            --this.a;
-            if (this.a > 0) {
-                return 0;
-            } else {
-                this.a += (60 + random.nextInt(60)) * 20;
-                if (world.c() < 5 && world.worldProvider.g()) {
-                    return 0;
-                } else {
-                    int i = 0;
-                    Iterator iterator = world.players.iterator();
-
-                    while (iterator.hasNext()) {
-                        EntityHuman entityhuman = (EntityHuman) iterator.next();
-
-                        if (!entityhuman.isSpectator()) {
-                            BlockPosition blockposition = new BlockPosition(entityhuman);
-
-                            if (!world.worldProvider.g() || blockposition.getY() >= world.getSeaLevel() && world.e(blockposition)) {
-                                DifficultyDamageScaler difficultydamagescaler = world.getDamageScaler(blockposition);
-
-                                if (difficultydamagescaler.a(random.nextFloat() * 3.0F)) {
-                                    ServerStatisticManager serverstatisticmanager = ((EntityPlayer) entityhuman).getStatisticManager();
-                                    int j = MathHelper.clamp(serverstatisticmanager.getStatisticValue(StatisticList.CUSTOM.b(StatisticList.TIME_SINCE_REST)), 1, Integer.MAX_VALUE);
-                                    boolean flag2 = true;
-
-                                    if (random.nextInt(j) >= 72000) {
-                                        BlockPosition blockposition1 = blockposition.up(20 + random.nextInt(15)).east(-10 + random.nextInt(21)).south(-10 + random.nextInt(21));
-                                        IBlockData iblockdata = world.getType(blockposition1);
-                                        Fluid fluid = world.b(blockposition1);
-
-                                        if (SpawnerCreature.a(iblockdata, fluid)) {
-                                            GroupDataEntity groupdataentity = null;
-                                            int k = 1 + random.nextInt(difficultydamagescaler.a().a() + 1);
-
-                                            for (int l = 0; l < k; ++l) {
-                                                EntityPhantom entityphantom = new EntityPhantom(world);
-
-                                                entityphantom.setPositionRotation(blockposition1, 0.0F, 0.0F);
-                                                groupdataentity = entityphantom.prepare(difficultydamagescaler, groupdataentity, (NBTTagCompound) null);
-                                                world.addEntity(entityphantom, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.NATURAL); // CraftBukkit
-                                            }
-
-                                            i += k;
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                    }
-
-                    return i;
-                }
-            }
-        }
+        return 0;
     }
 }
-- 
2.14.1

