--- /usr/lib/pypy3.9/concurrent/futures/process.py	2023-01-04 12:20:02.734998093 -0500
+++ /usr/lib/pypy3.9/concurrent/futures/process.py	2023-01-04 12:26:09.735070876 -0500
@@ -668,6 +668,9 @@
             _threads_wakeups[self._executor_manager_thread] = \
                 self._executor_manager_thread_wakeup

+    def _adjust_process_count(self):
+        self._adjust_process_count_cpython()
+
     def _adjust_process_count_cpython(self):
         # if there's an idle process, we don't need to spawn a new one.
         if self._idle_worker_semaphore.acquire(blocking=False):
